File: src/main/java/co/edu/uniremington/gabrieladumar/Mantenimiento/model/Usuario.java
package co.edu.uniremington.gabrieladumar.Mantenimiento.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "usuario")
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usu")
    private Integer idUsu;

    @Column(name = "nom_usu", length = 50, nullable = false)
    private String nomUsu;

    @Column(name = "ape_usu", length = 50, nullable = false)
    private String apeUsu;

    @Column(name = "correo_usu", length = 80)
    private String correoUsu;

    @Column(name = "cont_usu", length = 255, nullable = false)
    private String contUsu;

    // Relación con Perfil
    @ManyToOne
    @JoinColumn(name = "id_perfil")
    private Perfil perfil;

    // Relación con Cliente
    @ManyToOne
    @JoinColumn(name = "id_cliente")
    private Cliente cliente;

    // Relación con Técnico
    @ManyToOne
    @JoinColumn(name = "id_tec")
    private Tecnico tecnico;


    // =========================================================
    // GETTERS Y SETTERS
    // =========================================================

    public Integer getIdUsu() {
        return idUsu;
    }

    public void setIdUsu(Integer idUsu) {
        this.idUsu = idUsu;
    }

    public String getNomUsu() {
        return nomUsu;
    }

    public void setNomUsu(String nomUsu) {
        this.nomUsu = nomUsu;
    }

    public String getApeUsu() {
        return apeUsu;
    }

    public void setApeUsu(String apeUsu) {
        this.apeUsu = apeUsu;
    }

    public String getCorreoUsu() {
        return correoUsu;
    }

    public void setCorreoUsu(String correoUsu) {
        this.correoUsu = correoUsu;
    }

    public String getContUsu() {
        return contUsu;
    }

    public void setContUsu(String contUsu) {
        this.contUsu = contUsu;
    }

    public Perfil getPerfil() {
        return perfil;
    }

    public void setPerfil(Perfil perfil) {
        this.perfil = perfil;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public Tecnico getTecnico() {
        return tecnico;
    }

    public void setTecnico(Tecnico tecnico) {
        this.tecnico = tecnico;
    }
    
}

File: mantenimiento_app/lib/models/usuario.dart
class Usuario {
  final int idUsu;
  final String nomUsu;
  final String apeUsu;
  final String correoUsu;
  final String contUsu;

  final int idPerfil;
  final String nombrePerfil;

  final int? idCliente;

  // Datos del técnico
  final int? idTecnico;
  final String? nombreTecnico;
  final String? apellidoTecnico;
  final String? correoTecnico;

  Usuario({
    required this.idUsu,
    required this.nomUsu,
    required this.apeUsu,
    required this.correoUsu,
    required this.contUsu,
    required this.idPerfil,
    required this.nombrePerfil,
    this.idCliente,
    this.idTecnico,
    this.nombreTecnico,
    this.apellidoTecnico,
    this.correoTecnico,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    final perfil = json['perfil'];
    final cliente = json['cliente'];
    final tecnico = json['tecnico'];

    return Usuario(
      idUsu: json['idUsu'] ?? 0,

      nomUsu: json['nomUsu'] ?? '',

      apeUsu: json['apeUsu'] ?? '',

      correoUsu: json['correoUsu'] ?? '',

      contUsu: json['contUsu'] ?? '',

      // PERFIL
      idPerfil: perfil != null
          ? perfil['idPerfil'] ?? 0
          : 0,

      nombrePerfil: perfil != null
          ? perfil['nombrePerfil'] ?? ''
          : '',

      // CLIENTE
      idCliente: cliente != null
          ? cliente['idCliente']
          : null,

      // TÉCNICO
      idTecnico: tecnico != null
          ? tecnico['idTec']
          : null,

      nombreTecnico: tecnico != null
          ? tecnico['nomTec']
          : null,

      apellidoTecnico: tecnico != null
          ? tecnico['apeTec']
          : null,

      correoTecnico: tecnico != null
          ? tecnico['correoTec']
          : null,
    );
  }
}